core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ComponentParser.java

```diff
@@ -0,0 +1,290 @@
+package io.metadew.iesi.openapi;
+
+import io.metadew.iesi.connection.database.oracle.OracleDatabaseService;
+import io.metadew.iesi.metadata.definition.component.Component;
+import io.metadew.iesi.metadata.definition.component.ComponentParameter;
+import io.metadew.iesi.metadata.definition.component.ComponentVersion;
+import io.metadew.iesi.metadata.definition.component.key.ComponentKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentParameterKey;
+import io.metadew.iesi.metadata.definition.component.key.ComponentVersionKey;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.oas.models.parameters.RequestBody;
+import io.swagger.v3.oas.models.responses.ApiResponses;
+import io.swagger.v3.oas.models.security.SecurityRequirement;
+import io.swagger.v3.oas.models.security.SecurityScheme;
+
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ComponentParser {
+    private static ComponentParser INSTANCE;
+
+    private static Map<String, SecurityScheme> securitySchemeMap;
+    private static Long versionNumber;
+    private static String connectionName;
```

Comment:
Why store these parameters in a static class field? You can create local fields in the methods for this

Reply:
have you looked at this comment?

Reply:
Yes, I removed the statis keyword and I created these variables on top becuase they are used for every component and doesn"t change. Their values are assigned on the parse method and to avoid passing these value to multiple methods, I prefered to leavethem accessible everywhere

Reply:
not really. Think about the scenario where two users want to load a different OpenAPI configuration at the same time? Then the class variables will collide

Reply:
If they are not static, normall they won't collide, no ?